### ISSUE TITLE
Removal of Plan param from Subscription interface

### ISSUE THREAD

**Issue by pocketcolin** [stripe/stripe-node#1120]
I recently updated from `8.29.0` to `8.135.0` so bare with me as I know a lot has changed and I'm not sure exactly when this changed. In the `Subscriptions.d.ts` declaration file in version 8.29.0, Plan was defined as such:

```
   /** 
   * Hash describing the plan the customer is subscribed to. Only set if the subscription contains a single plan.
   */
      plan: Stripe.Plan | null;
```

As of 8.135.0, that prop is no longer on the `Subscription` interface. Based on my tests, the API does still return this property if the subscription contains a single plan. However, I haven't seen in any changelogs why this was removed and I noticed that `plan` is no longer listed as a return prop in the documentation for the [subscription update method](https://stripe.com/docs/api/subscriptions/update). Has this prop been deprecated?

I apologize in advance if this was mentioned in a changelog or discussed elsewhere and I missed it.

**Comment by remi-stripe** [comment]
@pocketcolin Thanks for reaching out! The `plan` property top-level on Subscription has been considered deprecated for a few years now which is why we recently removed it from our docs and from the type definitions. We shipped support for multiple prices per Subscription a few years ago and the `plan` property top-level didn't make sense in that world and was only set if you had only one price on that subscription. It's the same for the top-level `quantity` which also doesn't make sense when you use metered billing for example.

Similarly, we recently shipped the Price API to unify the previous Plan and SKU APIs. This means that the Plan resource is considered deprecated overall and we recommend 

Instead we recommend using `items.data[0].price` instead!

The API has not changed and we haven't removed this for now so it's mostly clean up in types and docs so that net new users do not get confused but properties they shouldn't rely on. If we remove this in the future it will be behind an API version since it would be a breaking change and it wouldn't affect existing integrations (except on the type signature verification in your code).

We don't directly document those changes well today unfortunately as it mostly disappears from docs and Typescript definitions. It's something we want to improve in the future by explicitly reporting deprecated properties in types possibly to help developers self-serve.

Let me know if that clarifies the issue!
